### ISSUE TITLE
keepCurrentZoomLevel not working

### ISSUE THREAD

**Issue by 1parkplace** [domoritz/leaflet-locatecontrol#102]
Any idea why? Is there anything conflicting in my options below?

``` js
L.control.locate({
            keepCurrentZoomLevel: true,
            follow: true,
            stopFollowingOnDrag: true,
            metric: false,
            strings: {
                title: "Find my Location",
                outsideMapBoundsMsg: "You seem to be located outside the bounds of this MLS Region."
            }
        }).addTo(this.map);
```

**Comment by 1parkplace** [comment]
Not sure if it's related by I get the following error in mapbox.js when I hit my locate buttons:

TypeError: t is undefined
https://api.tiles.mapbox.com/mapbox.js/v2.1.3/mapbox.js
Line 1

**Comment by domoritz** [comment]
It's working for me. Maybe it's a problem with mapbox. Reopen if still an issue.

**Comment by espy** [comment]
I have the same issue (it's a bit of an edge case), and I think I figured it out.

I just saw that `keepCurrentZoomLevel: true` overrides `locateOptions.maxZoom`, but what overrides everything are the [event bounds](https://github.com/domoritz/leaflet-locatecontrol/blob/gh-pages/src/L.Control.Locate.js#L168), which in my case are enormous, because the location accuracy is very very bad. As in, 30km radius bad.  

**So, in short: if the accuracy is very low, the event bounds are huge, and override any zoom setting.**

This is probably intentional, since you'd want to see the whole accuracy circle overlay after geolocation. Since this also overrides `locateOptions.maxZoom`, there doesn't seem to be a way to stay zoomed in if the GPS accuracy is bad. 

Maybe this helps someone :)

And thanks for the great plugin, @domoritz !

**Comment by domoritz** [comment]
That's an interesting edge case that I haven't thought about.

**Comment by ghost** [comment]
Hi. Do you found a fix for this issue?. Im using mapbox.js and in my site: www.mdqmap.com keepCurrentZoomLevel: true is not working too....

**Comment by domoritz** [comment]
No, I haven't looked at it. I am happy to review pull requests, though.

**Comment by domoritz** [comment]
I think I found it. In https://github.com/domoritz/leaflet-locatecontrol/blob/gh-pages/src/L.Control.Locate.js#L169 it should be `||`, not `%%`. I have to test it.

**Comment by domoritz** [comment]
@cesarGimenez @espy @1parkplace Should be fixed now. Let me know if you still see this issue.

**Comment by cesarGimenez** [comment]
Thanks! @domoritz ! but in my site (http://mdqmap.com/) I still have the issue. I set the options keepCurrentZoomLevel: true, drawCircle: false, but when user click the location button  zoom in too much on movile and zoom out on desktop, maybe I missing something....

**Comment by ghost** [comment]
Many thanks @domoritz, it works! but only if I use rawgit, not with the mapbox CDN...

https://www.mapbox.com/mapbox.js/plugins/#leaflet-locatecontrol

**Comment by domoritz** [comment]
@cesarGimenez I guess mapbox has to update its CDN.

**Comment by ghost** [comment]
@domoritz  can I use rawgit for production meanwhile?

**Comment by domoritz** [comment]
I highly recommend bower for anything serious. Also, I sent a pull request to update mapbox at https://github.com/mapbox/mapbox.js-plugins/pull/24
